packages/ui/src/components/breath-player/use-dot-animation.test.js

```diff
@@ -0,0 +1,92 @@
+import { renderHook, act } from '@testing-library/react-hooks'
+import { defaults } from './animation-steps'
+import { useDotAnimation } from './use-dot-animation'
+
+jest.mock('react-spring')
+// eslint-disable-next-line import-order-alphabetical/order
+const { useSpring } = require('react-spring')
+
+test('useDotAnimation should step though animation json', () => {
+  let onRest = null
+  const set = jest.fn().mockImplementation(({ onRest: r }) => {
+    onRest = r
+  })
+  set.id = 'foo'
+  const setCurrentStep = jest.fn()
+  const stop = jest.fn()
+  const props = {}
+  const json = [
+    { scale: [1.6, 1], outerScale: [1.7, 1], duration: [10, 10] },
+    { scale: [1.8, 1.1], outerScale: [1.9, 1.1], duration: [20, 20] },
+  ]
+  const onRenderEndCard = jest.fn()
+  useSpring.mockImplementation(() => [props, set, stop])
+  const { rerender } = renderHook(
+    ({ currentStep }) =>
+      useDotAnimation({
+        json,
+        isOpen: true,
+        setCurrentStep,
+        currentStep,
+        onFinish: jest.fn(),
+        onRenderEndCard,
+        EndCard: jest.fn().mockImplementation(() => null),
+        plays: 0,
+        setPlays: jest.fn(),
+      }),
+    { initialProps: { currentStep: 0 } }
+  )
+
+  const step1 = set.mock.calls[0][0]
```

Comment:
@itsjustmath the reason the test was failing was because of a compilation issue with some `eslint-ignore` comment. talk about irony